lectures/symbolic_sympy.md

```diff
@@ -0,0 +1,272 @@
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.5
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Introduction to Symbolic Algebra and Calculus with Sympy
+
+
+## Overview
+
+Symbolic computation, or computer algebra, involves using algorithms and software to perform exact computations and manipulate mathematical equations. While **Mathematica** is a popular tool for symbolic computations, it's proprietary and can be costly. As an open-source alternative, the **sympy** library in Python offers a comprehensive set of functionalities for symbolic mathematics. This allows users to perform a range of operations, from algebraic manipulations to calculus, providing exact solutions instead of numerical approximations.
+
+
+## Getting Started
+
+Let's get started with **sympy** by installing and setting up the package.
+
+```{code-cell} ipython3
+! pip install sympy
+```
+
+We'll start by importing the necessary **sympy** functions and initializing our environment.
+
+```{code-cell} ipython3
+from sympy import symbols, init_printing
+
+# Enable best printer available
+init_printing()
+
+# Let's define some symbols to work with
+x, y, z = symbols('x y z')
+```
+
+We can now use these symbols **x**, **y**, and **z** to build symbolic expressions.
+
+```{code-cell} ipython3
+---
+mystnb:
+  image:
+    width: 10%
+---
+# An expression
+expr = (x + y) ** 2
+expr
+```
+
+## Symbolic Algebra
+
+
+### Algebraic Expressions
+
+**sympy** provides several functions to create and manipulate algebraic expressions. Let's look at a few examples.
+
+Creating Algebraic Expressions
+
+```{code-cell} ipython3
+---
+mystnb:
+  image:
+    width: 17.5%
+---
+from sympy import Eq, solve
+
+# Create an equation
+eq = Eq(x**2 - 3*x + 2, 0)
+eq
+```
+
+Manipulating Algebraic Expressions
+
+```{code-cell} ipython3
+---
+mystnb:
+  image:
+    width: 10%
+---
+from sympy import simplify
+
+# Simplify an expression
+simplified_expr = simplify(expr)
+simplified_expr
+```
+
+### Solving Algebraic Equations
+
+We can solve equations using the **solve** function in **sympy**.
+
+```{code-cell} ipython3
+---
+mystnb:
+  image:
+    width: 5.5%
+---
+# Solve the equation
+sol = solve(eq, x)
+sol
+```
+
+## Symbolic Calculus
+
+**sympy** allows us to perform various calculus operations such as differentiation and integration.
+
+
+### Limits
+
+We can compute limits for a given function.
+
+```{code-cell} ipython3
+---
+mystnb:
+  image:
+    width: 3%
+---
+from sympy import limit
+
+# Define a function
+f = x ** 2 / (x - 1)
+
+# Compute the limit
+lim = limit(f, x, 1)
+lim
+```
+
+### Derivatives
+
+We can differentiate any **sympy** expression using **diff(func, var)**.
+
+```{code-cell} ipython3
+---
+mystnb:
+  image:
+    width: 20%
+---
+from sympy import diff
+
+# Differentiate a function
+df = diff(f, x)
+df
+```
+
+### Integrals
+
+We can compute definite and indefinite integrals.
```

Comment:
As mentioned above, please briefly introduce the function that you are using here.